src/api/app/controllers/concerns/webui/rescue_handler.rb

```diff
@@ -0,0 +1,56 @@
+module Webui::RescueHandler
+  extend ActiveSupport::Concern
+
+  included do
+    rescue_from Pundit::NotAuthorizedError do |exception|
+      pundit_action = case exception.try(:query).to_s
+                      when 'index?' then 'list'
+                      when 'show?' then 'view'
+                      when 'create?' then 'create'
+                      when 'new?' then 'create'
+                      when 'update?' then 'update'
+                      when 'edit?' then 'edit'
+                      when 'destroy?' then 'delete'
+                      when 'create_branch?' then 'create_branch'
+                      else exception.try(:query)
+                      end
+      message = if pundit_action && exception.record
+                  "Sorry, you are not authorized to #{pundit_action} this #{exception.record.class}."
+                else
+                  'Sorry, you are not authorized to perform this action.'
+                end
+      if request.xhr?
+        render json: { error: message }, status: 400
+      else
+        flash[:error] = message
+        redirect_back(fallback_location: root_path)
+      end
+    end
+
+    rescue_from Backend::Error, Timeout::Error do |exception|
+      Airbrake.notify(exception)
+      message = case exception
+                when Backend::Error
+                  'There has been an internal error. Please try again.'
+                when Timeout::Error
+                  'The request timed out. Please try again.'
+                end
+
+      if request.xhr?
+        render json: { error: message }, status: 400
+      else
+        flash[:error] = message
+        redirect_back(fallback_location: root_path)
+      end
+    end
+
+    # FIXME: This is more than stupid. Why do we tell the user that something isn't found
```

Comment:
What a lovely comment!

Reply:
not from me :rofl:

Reply:
i will drop it

Reply:
> not from me rofl

Yes I know! The point is valid, the tone is quite bad.

Reply:
fixed it.